### PR TITLE
feat!: adapt econCommitteeCharter to work with smart wallet continuing offers

### PR DESCRIPTION
**Pull request by dckc**
refs: #6598

## Description

The `psmCharter` had essentially nothing specific to the PSM; it's suitable for all param governance and pausing. So after pruning a few incidental details of the `psmCharter`, we use it in place of the previous `econCommitteeCharter`.

So now introducing the vault factory, reserve, amm and such is not done at charter contract creation time but rather using a creator facet method.

### Security Considerations

A bit more power is concentrated in the one charter contract.

### Documentation Considerations

The `--psmCharterAcceptOfferId` arg of `agops` works without changing its name, so I left it alone. Is that the right call?

### Testing Considerations

Existing tests seem to provide reasonable coverage.

### CODE REVIEW

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 1]
```diff
@@ -69,6 +68,14 @@ const MILLI = 1_000_000n;
  * }>} EconomyBootstrapSpace
  */
 
+/**
+ * @param {ERef<ZoeService>} zoe
+ * @param {Installation<import('../econCommitteeCharter').start>} inst
+ * @typedef {Awaited<ReturnType<typeof startCharterInstance>>} EconCharterStartResult
+ */
+// eslint-disable-next-line no-unused-vars
+const startCharterInstance = (zoe, inst) => E(zoe).startInstance(inst);
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/proposals/econ-behaviors.js:77]: Why define this if it's not used?

File: packages/inter-protocol/src/proposals/committee-proposal.js [hunk 2]
```diff
@@ -22,12 +22,17 @@ export const inviteCommitteeMembers = async (
...
     installation: {
       consume: { binaryVoteCounter: counterP },
     },
+    instance: {
+      consume: { amm, reserve, VaultFactory },
+    },
   },
   { options: { voterAddresses } },
 ) => {
+  // TODO: why doesn't this use the promise space?
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/proposals/committee-proposal.js]: This [isn't a TODO](https://github.com/Agoric/agoric-sdk/wiki/Coding-Style#:~:text=Code-,comments,-Try%20to%20write) (it doesn't recommend an action, and doesn't deserve an issue.)

The wiki recommends possibly `FIXME`.

**Review by Chris-Hibbert — APPROVED** [review]
I didn't remember that I had written the charter so generically. Good on you for recognizing that!

Looks fine to me.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 1 above)
Comment (dckc) [reply packages/inter-protocol/src/proposals/econ-behaviors.js:77]: For its return type, `EconCharterStartResult`.

Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js:77]: would calling it `_startCharterInstance` obviate the need for the disable comment?

[Not a blocking suggestion.]

Comment (dckc) [reply packages/inter-protocol/src/proposals/econ-behaviors.js:77]: no; then we run into eslint `no-underscore-dangle`

Comment (turadg) [reply packages/inter-protocol/src/proposals/econ-behaviors.js:77]: https://github.com/Agoric/agoric-sdk/pull/6614

**Comment by turadg** [comment]
> The --psmCharterAcceptOfferId arg of agops works without changing its name, so I left it alone. Is that the right call?

I think that's fine. It's still the charter for the PSM even if the contract is econCommitteeCharter.

For testing considerations I would have requested a manual run through `agops-governance-smoketest.sh`. I just did and ran into some problems. I don't think they're caused by this, but another recent bootstrap change around denoms,
```
2022-12-02T15:43:06.965Z SwingSet: vat: v17: walletFactory.fromBridge: wallet { blockHeight: 251, blockTime: 1669995781, owner: 'agoric109q3uc0xt8aavne94rgd6rfeucavrx924e0ztf', spendAction: '{"body":"{\\"method\\":\\"executeOffer\\",\\"offer\\":{\\"id\\":1669995785517,\\"invitationSpec\\":{\\"source\\":\\"contract\\",\\"instance\\":{\\"@qclass\\":\\"slot\\",\\"index\\":0},\\"publicInvitationMaker\\":\\"makeWantMintedInvitation\\"},\\"proposal\\":{\\"give\\":{\\"In\\":{\\"brand\\":{\\"@qclass\\":\\"slot\\",\\"index\\":1},\\"value\\":{\\"@qclass\\":\\"bigint\\",\\"digits\\":\\"10002\\"}}},\\"want\\":{\\"Out\\":{\\"brand\\":{\\"@qclass\\":\\"slot\\",\\"index\\":2},\\"value\\":{\\"@qclass\\":\\"bigint\\",\\"digits\\":\\"10000\\"}}}}}}","slots":["board06120","board0566","board05311"]}', type: 'WALLET_SPEND_ACTION' }
2022-12-02T15:43:06.965Z SwingSet: vat: v17: walletFactory: { wallet: Object [Alleged: SmartWallet self] {}, actionCapData: { body: '{"method":"executeOffer","offer":{"id":1669995785517,"invitationSpec":{"source":"contract","instance":{"@qclass":"slot","index":0},"publicInvitationMaker":"makeWantMintedInvitation"},"proposal":{"give":{"In":{"brand":{"@qclass":"slot","index":1},"value":{"@qclass":"bigint","digits":"10002"}}},"want":{"Out":{"brand":{"@qclass":"slot","index":2},"value":{"@qclass":"bigint","digits":"10000"}}}}}}', slots: [ 'board06120', 'board0566', 'board05311' ] } }
2022-12-02T15:43:06.978Z SwingSet: vat: v17: wallet agoric109q3uc0xt8aavne94rgd6rfeucavrx924e0ztf starting executeOffer 1669995785517
2022-12-02T15:43:07.037Z SwingSet: ls: v1: Logging sent error stack (Error#2)
2022-12-02T15:43:07.037Z SwingSet: ls: v1: Error#2: cannot grab 10002ibc/toyellie coins: 0ibc/toyellie is smaller than 10002ibc/toyellie: insufficient funds [agoric-labs/cosmos-sdk@v0.45.4-alpha.agoric.2/x/bank/keeper/send.go:187]
2022-12-02T15:43:07.037Z SwingSet: ls: v1: Error: cannot grab 10002ibc/toyellie coins: 0ibc/toyellie is smaller than 10002ibc/toyellie: insufficient funds [agoric-labs/cosmos-sdk@v0.45.4-alpha.agoric.2/x/bank/keeper/send.go:187]
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:54)
```
(there was a bug in sending previousOfferId too, which I've rolled into https://github.com/Agoric/agoric-sdk/pull/6615 )

**Pull request merged**